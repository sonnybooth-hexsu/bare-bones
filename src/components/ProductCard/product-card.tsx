import React from "react"

type ProductCardProps = {
  title: string
  path: string
  excerpt: string
  image: string
}

export const ProductCard = ({
  title,
  path,
  excerpt,
  image,
}: ProductCardProps) => {
  return (
    <div className="bg-white">
      <div className="bg-red-200 pt-16 pb-16 flex justify-center">
        <img className="h-20" src={image} alt="Lips" />
      </div>
      <div className="pr-8 pb-8 pl-8">
        <h2
          data-testid="productCardTitle"
          className="mt-12 mb-2 text-3xl tracking-wide"
        >
          {title}
        </h2>
        <p className="mb-12">{excerpt}</p>
        <a href={path}>View Product</a>
      </div>
    </div>
  )
}
