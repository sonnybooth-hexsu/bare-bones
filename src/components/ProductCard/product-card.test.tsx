/* eslint no-undef: 0 */

import React from "react"
import { render, cleanup } from "@testing-library/react"
import { getByText, getByTestId } from "@testing-library/dom"
import { ProductCard } from "./product-card"

afterEach(() => {
  cleanup()
})

describe("<ProductCard />", () => {
  it("should match snapshot", () => {
    const { container } = render(
      <ProductCard
        title="Test title"
        path="/test-product"
        excerpt="I am a test excerpt"
        image="/test-product.jpg"
      />
    )

    expect(container).toMatchSnapshot()
  })

  it("should have the title text passed through", () => {
    const { container } = render(
      <ProductCard
        title="Test title"
        path="/test-product"
        excerpt="I am a test excerpt"
        image="/test-product.jpg"
      />
    )

    const titleEl = getByTestId(container, "productCardTitle")

    expect(getByText(titleEl, "Test title"))
  })
})
