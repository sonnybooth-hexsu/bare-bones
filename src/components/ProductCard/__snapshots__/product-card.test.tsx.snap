// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<ProductCard /> should match snapshot 1`] = `
<div>
  <div
    class="bg-white"
  >
    <div
      class="bg-red-200 pt-16 pb-16 flex justify-center"
    >
      <img
        alt="Lips"
        class="h-20"
        src="//test-product.jpg"
      />
    </div>
    <div
      class="pr-8 pb-8 pl-8"
    >
      <h2
        class="mt-12 mb-2 text-3xl tracking-wide"
        data-testid="productCardTitle"
      >
        Test title
      </h2>
      <p
        class="mb-12"
      >
        I am a test excerpt
      </p>
      <a
        href="/test-product"
      >
        View Product
      </a>
    </div>
  </div>
</div>
`;
